### DISCUSSION TITLE
possible bug with the configure file in meep version 1.24.0

### DISCUSSION THREAD

**Discussion by gpourtois** [NanoComp/meep#2232] [Q&A]
Dear meep developers,

I would like to draw your attention to a possible issue with the configuration file of meep, version 1.24.0.
I'm trying to compile it on red hat linux 8.6. 

In the configuration step, the configure script returns: "checking for libctl dir... ./configure: line 29448: $'\357\277\274': command not found". It seems that the '\357\277\274': error is linked to some file encoding issues within the configure file and that it contains one or several hidden characters at line 29448.

Editing the file with vim and checking line 29448 does not display anything odd:
"29448      LIBCTL_DIR="$LIBCTL_DIR/share/libctl""

However, removing the first 'white space' character of line 29448 (command x in vim) returns the following
"29448 ��    LIBCTL_DIR="$LIBCTL_DIR/share/libctl""

If I then remove the 2 � characters and restart the configuration process, the step processes without any issue.

Did someone else meet the same issue or is this something specific to my linux settings?
Could someone check this and to amend the configure file if you can reproduce the issue?
I'm unfortunately not literate enough with git to push the modified configure file.

Thx a lot,

Regards,
Geoffrey

**Comment by smartalecH** [comment]
Hmm I can't replicate this on an M1 Mac or on an Ubuntu system. It compiles just fine for me.

**Comment by Yaraslaut** [comment]
I can not reproduce this issue on RHEL 8.6, you can try to update configure script via: `autoreconf --verbose --install --symlink --force`

**Reply by smartalecH** [reply]
You may need to run it a couple of times too

**Reply by gpourtois** [reply]
Dear all,

Thank you for your help and efforts. 

Out of curiosity, I also tried older versions of meep (1.20 and 1.16) and I went through the same issue with the same  LIBCTL_DIR="$LIBCTL_DIR/share/libctl" library line. The trick explained in my first post fixed the issue for me. 
Oddly enough, when I installed meep on an ubuntu system, I did not face the issue. An important thing is that for the ubuntu system I used the 'standard' installation process, while the RHEL 8.6 version I have access to is a stripped version that the IT department of my institution installed. 

I'm suspecting that one (or several) rpm packages is/are missing that cover(s) the encoding of the problematic character.
However, I haven't yet figured out which one(s).

**Comment by stevengj** [comment]
Do you have a directory name that includes some non-ASCII characters, which might be confusing the shell?

**Reply by gpourtois** [reply]
Thank you for your suggestion; this could explain the issue. 
Unfortunately, I installed it in /data/other/measy/manual_install/meep-1.24.0/, which is a posix compliant directory path structure. I don't see anything that would stand out in the directory name.
